vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion.cc

```diff
@@ -256,44 +286,81 @@ void Promotion::OnCompletedAttestation(
     ledger::PromotionPtr promotion,
     ledger::AttestPromotionCallback callback) {
   if (!promotion) {
+    BLOG(ledger_, ledger::LogLevel::LOG_ERROR)
+    << "Promotion does not exist ";
     callback(ledger::Result::LEDGER_ERROR, nullptr);
     return;
   }
 
   if (promotion->status == ledger::PromotionStatus::CLAIMED) {
+    BLOG(ledger_, ledger::LogLevel::LOG_ERROR) << "Promotions already claimed";
     callback(ledger::Result::GRANT_ALREADY_CLAIMED, nullptr);
     return;
   }
 
   promotion->status = ledger::PromotionStatus::ATTESTED;
-  ledger_->InsertOrUpdatePromotion(
-        promotion->Clone(),
-        [](const ledger::Result _){});
 
-  auto claim_callback = std::bind(&Promotion::Complete,
+  auto save_callback = std::bind(&Promotion::AttestedSaved,
       this,
       _1,
       braveledger_bind_util::FromPromotionToString(promotion->Clone()),
       callback);
 
-  ClaimTokens(std::move(promotion), claim_callback);
+  ledger_->InsertOrUpdatePromotion(promotion->Clone(), save_callback);
 }
 
-void Promotion::Complete(
+void Promotion::AttestedSaved(
     const ledger::Result result,
     const std::string& promotion_string,
     ledger::AttestPromotionCallback callback) {
+  if (result != ledger::Result::LEDGER_OK) {
+    BLOG(ledger_, ledger::LogLevel::LOG_ERROR) << "Save failed ";
+    callback(ledger::Result::LEDGER_ERROR, nullptr);
```

Comment:
Wouldn't we want to pass back `result` here? It might be more specific/useful than `LEDGER_ERROR`.

Reply:
done